packages/autocomplete-plugin-recent-searches/src/usecases/localStorage/search.ts

```diff
@@ -0,0 +1,21 @@
+import { RecentSearchesItem } from '../../types';
+
+export type SearchParams<TItem> = {
+  query: string;
+  items: TItem[];
+  limit: number;
+};
+
+export function search<TItem extends RecentSearchesItem>({
+  query,
+  items,
+  limit,
+}: SearchParams<TItem>) {
+  if (!query) {
+    return items.slice(0, limit);
+  }
+
+  return items
+    .filter((item) => item.query.toLowerCase().startsWith(query.toLowerCase()))
```

Comment:
do we polyfill startsWith or mention browser support somewhere?

Reply:
We'll be explicit about browser support in the doc.

FYI [Microsoft Teams stops supporting IE11 at the end of this month](https://techcommunity.microsoft.com/t5/microsoft-365-blog/microsoft-365-apps-say-farewell-to-internet-explorer-11-and/ba-p/1591666) (November 2020).